include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -104,65 +104,77 @@ cigar_op compare_aligned_values(reference_char_type const reference_char,
+ *                         second (query) sequence.
  * \param  query_start_pos The start position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  query_end_pos   The end position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  extended_cigar  Whether to print the extended cigar alphabet or not. See CIGAR operation.
- * \returns An std::string representing the alignment as a CIGAR string.
+ * \returns An std::vector<seqan3::cigar> representing the alignment.
  *
- * ### Theoretical Example:
+ * \details
  *
- * The following alignment reference sequence on top and the query sequence at
- * the bottom.
+ * \attention Note that CIGAR elements (respectively by their
+ *            CIGAR operation) are always related to one of the two sequences
+ *            in a pairwise alignment. In this case, the resulting cigar_vector
+ *            is based on sequence at the second position of the \p alignment pair,
+ *            namely the query sequence.
+ *
+ * ### Example:
+ *
+ * Given the following alignment reference sequence on top and the query sequence at
+ * the bottom:
  * ```
  * ATGG--CGTAGAGC
  * |||X  |||X|  |
  * ATGCCCCGTTG--C
  * ```
- * In this case, the function seqan3::detail::get_cigar_string will return
- * the following cigar string when printed: "4M2I5M2D1M". The extended cigar
- * string would look like this: "3=1X2I3=1X1=2D1=".
- * \sa seqan3::aligned_sequence
+ * In this case, the function seqan3::detail::get_cigar_vector will return
+ * the following cigar vector: "[('M',4),('I',2),('M',5),('D',2),('M',1)]".
+ * The extended cigar string would look like this: "[('=',3)('X',1)('I',2)('=',3)('X',1)('=',1)('D',2)('=',1)]".
+ *
+ * \include test/snippet/io/alignment_file/get_cigar_vector.cpp
+ *
+ * \sa seqan3::AlignedSequence
  */
-template <std::ranges::forward_range ref_seq_type, std::ranges::forward_range query_seq_type>
+template <tuple_like alignment_type>
 //!\cond
-    requires std::detail::weakly_equality_comparable_with<gap, reference_t<ref_seq_type>> &&
-             std::detail::weakly_equality_comparable_with<gap, reference_t<query_seq_type>>
+    requires std::tuple_size_v<remove_cvref_t<alignment_type>> == 2
 //!\endcond
-std::string get_cigar_string(ref_seq_type && ref_seq,
-                             query_seq_type && query_seq,
-                             uint32_t const query_start_pos = 0,
-                             uint32_t const query_end_pos = 0,
-                             bool const extended_cigar = false)
+std::vector<cigar> get_cigar_vector(alignment_type && alignment,
+                                    uint32_t const query_start_pos = 0,
+                                    uint32_t const query_end_pos = 0,
+                                    bool const extended_cigar = false)
 {
+    auto & ref_seq = get<0>(alignment);
```

Comment:
Actually I have to correct my self. std::get is not a customisation point. So you would not be allowed to overload get in std namespace. I think we did it in some places but it must be removed. This also means, that you don't need `using std::get` because you would not look for an overload in std namespace for a third party type.

Reply:
But I need it if I want to allow types that allow either `std::get` access or `seqan3::get` access.

Reply:
yes, you are right. Confusing topic

Reply:
But is it added now? The block does not seem updated?